### PR TITLE
Github io docs sort by client, mgmt, track2, track1

### PR DESCRIPTION
**Pull request by sima-zhu**
Issue: https://github.com/Azure/azure-sdk/issues/3282

Current look:
https://azure.github.io/azure-sdk-for-net/keyvault.html

After change:
image

### CODE REVIEW

**Comment by sima-zhu** [comment]
https://github.com/Azure/azure-sdk/issues/3282

File: eng/common/docgeneration/Generate-DocIndex.ps1
Comment (weshaggard) [inline eng/common/docgeneration/Generate-DocIndex.ps1:86]:
```suggestion
        $isNew = $packageInfo[0].New -eq 'true'
```
I like to avoid track1/track2 code names.

File: eng/common/docgeneration/Generate-DocIndex.ps1
Comment (weshaggard) [inline eng/common/docgeneration/Generate-DocIndex.ps1:88]: Lets sort by new then client so that we also get the track 2 mgmt higher then the track 1 client or mgmt libraries.

Comment (weshaggard) [reply eng/common/docgeneration/Generate-DocIndex.ps1:88]:
```suggestion
        $orderServiceMapping[$artifact] = [PSCustomObject][ordered]@{
          New = $packageInfo[0].New
          Type = $packageInfo[0].Type
          ServiceName = $serviceName
          DisplayName =$packageInfo[0].DisplayName.Trim()
       }
```

File: eng/common/docgeneration/Generate-DocIndex.ps1
Comment (weshaggard) [inline eng/common/docgeneration/Generate-DocIndex.ps1:118]: Once we use custom objects above the Sort-Object can pass things by Name.
```suggestion
    foreach ($serviceMapping in ($tocContent.GetEnumerator() | Sort-Object New, Type, ServiceName, DisplayName)) {
```

File: eng/common/docgeneration/Generate-DocIndex.ps1
Comment (weshaggard) [inline eng/common/docgeneration/Generate-DocIndex.ps1:121]:
```suggestion
        $serviceName = $serviceMapping.ServiceName
        $displayName = $serviceMapping.DisplayName
```

File: eng/common/docgeneration/Generate-DocIndex.ps1
Comment (danieljurek) [inline eng/common/docgeneration/Generate-DocIndex.ps1:127]: I think `sortedToc` needs a `$` in front.

File: eng/common/docgeneration/Generate-DocIndex.ps1
Comment (weshaggard) [inline eng/common/docgeneration/Generate-DocIndex.ps1:121]:
```suggestion
    $sortedToc = $tocContent.GetEnumerator() | Sort-Object {
```

File: eng/common/docgeneration/Generate-DocIndex.ps1
Comment (weshaggard) [reply eng/common/docgeneration/Generate-DocIndex.ps1:87]: Or if you want to keep this condition you should change the name of this property from New because it doesn't represent the correct value.

**Comment by weshaggard** [comment]
Lets please wait until after release week before merging this.

File: eng/common/docgeneration/Generate-DocIndex.ps1
Comment (sima-zhu) [reply eng/common/docgeneration/Generate-DocIndex.ps1:87]: Based on my experiment, $False comes before $True.

File: eng/common/docgeneration/Generate-DocIndex.ps1
Comment (sima-zhu) [reply eng/common/docgeneration/Generate-DocIndex.ps1:88]: Then we need to define the order, 
New has three types: true, false, empty
Type has three types: client, mgmt, empty.

Don't want to waste code to define the order.

File: eng/common/docgeneration/Generate-DocIndex.ps1
Comment (weshaggard) [inline eng/common/docgeneration/Generate-DocIndex.ps1:125]: I suggest doing this mapping in the custom object creation above.

File: eng/common/docgeneration/Generate-DocIndex.ps1
Comment (weshaggard) [inline eng/common/docgeneration/Generate-DocIndex.ps1:91]: I suggest adding Artifact as a property here as well because that will make sorting a little easier. See my comment below on what I suggest there.

File: eng/common/docgeneration/Generate-DocIndex.ps1
Comment (weshaggard) [inline eng/common/docgeneration/Generate-DocIndex.ps1:129]: This sorting looks a little more complex then it should be. I suggest something like:
```suggestion
    $sortedToc = $tocContent.Values | Sort-Object New, Type, ServiceName, DisplayName, Artifact
```

Of course this means you need to add Artifact to the property set of the custom object. Then you can change the below code to this as well. 

```
        $artifact = $serviceMapping.Artifact
        $serviceName = $serviceMapping.ServiceName
        $displayName = $serviceMapping.DisplayName
```

Comment (sima-zhu) [reply eng/common/docgeneration/Generate-DocIndex.ps1:129]: It is not there. We do not loop into the mapping yet when sort this.

Comment (weshaggard) [reply eng/common/docgeneration/Generate-DocIndex.ps1]: I'm not sure I understand your comment here? If you do $tocContents.Values.

File: eng/common/docgeneration/Generate-DocIndex.ps1
Comment (sima-zhu) [reply eng/common/docgeneration/Generate-DocIndex.ps1:103]: $artifact is key of the map. We do sort on this as shown below.
```
  $sortedToc = $tocContent.GetEnumerator() | Sort-Object {
        $_.Value.NewIndex, 
        $_.Value.TypeIndex, 
        $_.Value.ServiceName, 
        $_.Value.DisplayName,
        $_.Key} // This is the artifact
```

File: eng/common/docgeneration/Generate-DocIndex.ps1
Comment (sima-zhu) [reply eng/common/docgeneration/Generate-DocIndex.ps1]: I thought we asked me define artifact right here. Sort by key means sort by artifact.

File: eng/common/docgeneration/Generate-DocIndex.ps1
Comment (weshaggard) [inline eng/common/docgeneration/Generate-DocIndex.ps1:140]:
```suggestion
     $sortedToc = $tocContent.Values | Sort-Object New, Type, ServiceName, DisplayName, Artifact
```

File: eng/common/docgeneration/Generate-DocIndex.ps1
Comment (weshaggard) [inline eng/common/docgeneration/Generate-DocIndex.ps1:142]:
```suggestion
        $artifact = $serviceMapping.Artifact
        $serviceName = $serviceMapping.ServiceName
        $displayName = $serviceMapping.DisplayName
```

File: eng/common/docgeneration/Generate-DocIndex.ps1
Comment (weshaggard) [inline eng/common/docgeneration/Generate-DocIndex.ps1:105]:
```suggestion
        $orderServiceMapping[$artifact] = [PSCustomObject][ordered]@{
            NewIndex = $newIndex
            TypeIndex = $typeIndex
            ServiceName = $serviceName
            DisplayName = $packageInfo[0].DisplayName.Trim()
            Artifact = $artifact
       }
```

File: eng/common/docgeneration/Generate-DocIndex.ps1
Comment (weshaggard) [inline eng/common/docgeneration/Generate-DocIndex.ps1:93]:
```suggestion
        $newIndex = $CustomOrder_New.IndexOf($packageInfo[0].New.ToLower())
        $newIndex = $newIndex -eq -1 ?  $CustomOrder_New.Count : $newIndex
```

File: eng/common/docgeneration/Generate-DocIndex.ps1
Comment (weshaggard) [inline eng/common/docgeneration/Generate-DocIndex.ps1:135]:
```suggestion
    $sortedToc = $tocContent.Values | Sort-Object New, Type, ServiceName, DisplayName, Artifact
```

**Review by weshaggard — APPROVED** [review]

**Pull request merged**
